Structurize crash when rotating decoration

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Are you using the latest Structurize Version?

- [x] I am running the latest **beta/release** version of Structurize for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.


### Did you check on the Wiki? or ask on Discord?

- [x] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/).

### Minecraft Version

1.21

### Structurize Version

1.0.802

### MineColonies Version (if related bug)

1.1.1242

### Related Mods and their Versions

Blockui 1.0.208
Neoforge 21.1.217
Biomes O Plenty 21.1.0.13 (note, BoP items are not used in the exact schematic that is crashing, but the style requires this mod, or it will be hidden)
Ars Nouveau 5.11.1
Archwood Collective stylepack from https://www.planetminecraft.com/project/archwood-collective-minecolonies-style-1-21/  (version is 19)

### Current Behavior

Crash when rotating decoration schematic.

### Expected Behavior

No crash.

### Reproduction Steps

1. Create a colony.
2. Ensure you have a functioning builder (probably not needed for the repro)
3. Ensure you have Archwood Collective 1.21.1 style installed. This requires BoP and Ars nouveau. Versions of those mods are listed in related mods.
4. Using build tool (I was in creative), inside the colony, bring up preview of decorations/sourcegeneration/berryburntile.
5. Rotate (or mirror, according to another user). I did it with the keystrokes, I assume it'll bug for any rotation.
6. Crash

### Logs

https://gist.github.com/Shadizar/296959ac2078d51565f8676734fdc4fa

### Anything else?

[21Jan2026 17:34:04.816] [Render thread/ERROR] [net.minecraft.client.Minecraft/FATAL]: Reported exception thrown!
net.minecraft.ReportedException: KeyPressed event for BO screen
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:456) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.KeyboardHandler.keyPress(KeyboardHandler.java:407) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.KeyboardHandler.lambda$setup$8(KeyboardHandler.java:515) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:98) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.KeyboardHandler.lambda$setup$9(KeyboardHandler.java:515) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFWKeyCallbackI.callback(GLFWKeyCallbackI.java:44) ~[lwjgl-glfw-3.3.3.jar%23173!/:build 5]
	at MC-BOOTSTRAP/org.lwjgl@3.3.3+5/org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar%23185!/:build 5]
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3509) ~[lwjgl-glfw-3.3.3.jar%23173!/:build 5]
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:162) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.runTick(Minecraft.java:1220) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.run(Minecraft.java:807) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.main.Main.main(Main.java:230) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:32) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.5.jar%23112!/:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.component.CapturedBlock.applyRotationMirror(CapturedBlock.java:81) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.blueprints.v1.Blueprint.setRotationMirrorRelative(Blueprint.java:727) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.blueprints.v1.Blueprint.setRotationMirror(Blueprint.java:640) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.storage.rendering.types.BlueprintPreviewData.applyRotationMirrorAndSync(BlueprintPreviewData.java:329) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.storage.rendering.types.BlueprintPreviewData.rotate(BlueprintPreviewData.java:244) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.client.gui.AbstractBlueprintManipulationWindow.rotateRightClicked(AbstractBlueprintManipulationWindow.java:523) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.client.gui.AbstractBlueprintManipulationWindow.onUnhandledKeyTyped(AbstractBlueprintManipulationWindow.java:301) ~[structurize-1.0.802-1.21.1-snapshot.jar%23989!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.BOWindow.onKeyTyped(BOWindow.java:213) ~[blockui-1.0.208-1.21.1-snapshot.jar%23665!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.BOScreen.keyPressed(BOScreen.java:154) ~[blockui-1.0.208-1.21.1-snapshot.jar%23665!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.KeyboardHandler.lambda$keyPress$5(KeyboardHandler.java:411) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:451) ~[client-1.21.1-20240808.144430-srg.jar%23618!/:?]
	... 26 more
[21Jan2026 17:34:05.703] [Server thread/INFO] [MapFrontiers/]: ServerStoppingEvent done

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

In a minimal instance this still occurs


[21Jan2026 18:16:41.197] [Render thread/INFO] [net.minecraft.advancements.AdvancementTree/]: Loaded 332 advancements
[21Jan2026 18:18:22.166] [Server thread/INFO] [minecolonies/]: New Colony Id: 1 by 
[21Jan2026 18:18:22.179] [Render thread/INFO] [net.minecraft.client.gui.components.ChatComponent/]: [System] [CHAT] All land within 1 chunks of townhall has been claimed.
[21Jan2026 18:18:22.207] [Render thread/INFO] [net.minecraft.client.gui.components.ChatComponent/]: [System] [CHAT] News spreads swiftly of the newly founded settlement. Before long, pioneers from distant lands will flock to join the ranks of your burgeoning colony.
[21Jan2026 18:18:55.024] [Render thread/INFO] [net.minecraft.client.gui.components.ChatComponent/]: [System] [CHAT] All land within 1 chunks of builder has been claimed.
[21Jan2026 18:18:55.154] [Render thread/INFO] [net.minecraft.client.gui.components.ChatComponent/]: [System] [CHAT] All land within 2 chunks of builder has been claimed.
[21Jan2026 18:19:33.665] [Render thread/ERROR] [net.minecraft.client.Minecraft/FATAL]: Reported exception thrown!
net.minecraft.ReportedException: MousePressed event for BO screen
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:456) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.onPress(MouseHandler.java:95) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$setup$4(MouseHandler.java:202) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:98) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$setup$5(MouseHandler.java:202) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.3.jar%23173!/:build 5]
	at MC-BOOTSTRAP/org.lwjgl@3.3.3+5/org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar%23185!/:build 5]
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3438) ~[lwjgl-glfw-3.3.3.jar%23173!/:build 5]
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:134) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:143) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.platform.Window.updateDisplay(Window.java:292) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.runTick(Minecraft.java:1217) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.run(Minecraft.java:807) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.main.Main.main(Main.java:230) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:32) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.42.jar%23107!/:4.0]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.5.jar%23112!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.5.jar%23112!/:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.component.CapturedBlock.applyRotationMirror(CapturedBlock.java:81) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.blueprints.v1.Blueprint.setRotationMirrorRelative(Blueprint.java:727) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.blueprints.v1.Blueprint.setRotationMirror(Blueprint.java:640) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.storage.rendering.types.BlueprintPreviewData.applyRotationMirrorAndSync(BlueprintPreviewData.java:329) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.storage.rendering.types.BlueprintPreviewData.setBlueprint(BlueprintPreviewData.java:222) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.client.gui.WindowExtendedBuildTool.setBlueprint(WindowExtendedBuildTool.java:1144) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.client.gui.WindowExtendedBuildTool.handleBlueprintCategory(WindowExtendedBuildTool.java:1069) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/structurize@1.0.802-1.21.1-snapshot/com.ldtteam.structurize.client.gui.WindowExtendedBuildTool.onButtonClicked(WindowExtendedBuildTool.java:1024) ~[structurize-1.0.802-1.21.1-snapshot.jar%23211!/:1.0.802-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.controls.Button.handleClick(Button.java:115) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.Pane.click(Pane.java:597) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseEventProcessor(View.java:438) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseClickableEventHandler(View.java:391) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.click(View.java:234) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseEventProcessor(View.java:438) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.ScrollingContainer.mouseEventProcessor(ScrollingContainer.java:168) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseClickableEventHandler(View.java:391) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.click(View.java:234) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseEventProcessor(View.java:438) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseClickableEventHandler(View.java:391) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.click(View.java:234) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseEventProcessor(View.java:438) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.mouseClickableEventHandler(View.java:391) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.views.View.click(View.java:234) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/blockui@1.0.208-1.21.1-snapshot/com.ldtteam.blockui.BOScreen.mouseClicked(BOScreen.java:196) ~[blockui-1.0.208-1.21.1-snapshot.jar%23204!/:1.0.208-1.21.1-snapshot]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$onPress$0(MouseHandler.java:98) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:451) ~[client-1.21.1-20240808.144430-srg.jar%23200!/:?]
	... 28 more
[21Jan2026 18:19:33.888] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Stopping server

--- a/src/main/java/com/ldtteam/structurize/component/CapturedBlock.java
+++ b/src/main/java/com/ldtteam/structurize/component/CapturedBlock.java
@@ -67,21 +67,39 @@ public record CapturedBlock(BlockState blockState, Optional<CompoundTag> seriali
      */
     public CapturedBlock applyRotationMirror(final RotationMirror rotationMirror, final Level level)
     {
+        final BlockState rotatedState = rotationMirror.applyToBlockState(blockState);
+
+        // No BE data: just rotate the state.
         if (serializedBE.isEmpty())
         {
-            return new CapturedBlock(rotationMirror.applyToBlockState(blockState), serializedBE, itemStack);
+            return new CapturedBlock(rotatedState, serializedBE, itemStack);
         }
 
+        // If the rotated state cannot host a BE, drop any BE tag (prevents renderer/loader issues).
+        if (!rotatedState.hasBlockEntity())
+        {
+            return new CapturedBlock(rotatedState, Optional.empty(), itemStack);
+        }
+
+        // Rotate/migrate the BE tag using the Blueprint rotation logic
         final Blueprint blueprint = new Blueprint((short) 1, (short) 1, (short) 1, level.registryAccess());
-        blueprint.addBlockState(BlockPos.ZERO, blockState);
+        blueprint.addBlockState(BlockPos.ZERO, rotatedState);
         blueprint.getTileEntities()[0][0][0] = serializedBE.get();
         blueprint.setCachePrimaryOffset(BlockPos.ZERO);
         blueprint.setRotationMirrorRelative(rotationMirror, level);
 
-        return new CapturedBlock(blueprint.getPalette()[blueprint.getPalleteSize()],
-            Optional.of(blueprint.getTileEntities()[0][0][0]),
-            itemStack);
+        final CompoundTag rotatedTag = blueprint.getTileEntities()[0][0][0];
+        Optional<CompoundTag> beTag = Optional.ofNullable(rotatedTag);
+
+        // Drop invalid/empty BE tags.
+        if (beTag.isEmpty() || beTag.get().isEmpty() || !beTag.get().contains("id"))
+        {
+            beTag = Optional.empty();
+        }
+
+        return new CapturedBlock(rotatedState, beTag, itemStack);
     }
+
 
     public boolean hasBlockEntity()
     {

--- a/src/main/java/com/ldtteam/structurize/component/CapturedBlock.java
+++ b/src/main/java/com/ldtteam/structurize/component/CapturedBlock.java
@@ -1,5 +1,6 @@
 package com.ldtteam.structurize.component;
 
+import com.ldtteam.structurize.api.Log;
 import com.ldtteam.structurize.api.RotationMirror;
 import com.ldtteam.structurize.blueprints.v1.Blueprint;
 import com.mojang.serialization.Codec;
@@ -75,9 +76,10 @@ public record CapturedBlock(BlockState blockState, Optional<CompoundTag> seriali
             return new CapturedBlock(rotatedState, serializedBE, itemStack);
         }
 
-        // If the rotated state cannot host a BE, drop any BE tag (prevents renderer/loader issues).
+        // If the rotated state does not host a BE, drop any BE tag (prevents renderer/loader issues).
         if (!rotatedState.hasBlockEntity())
         {
+             Log.getLogger().warn("Block {} is not empty, but has no block entity.", rotatedState);
             return new CapturedBlock(rotatedState, Optional.empty(), itemStack);
         }
 

